specs.md

```diff
@@ -19,50 +19,140 @@ This approach is based on data gathered from public automatically resetted netwo
+         generate new_genesis 
+         return genesis_from_database(new_genesis)
+      else:
+         return genesis_from_database()
+
+The choice of period can be very important and have certain implications on netowrk stability and overall user experience. 
+
+For example shorter periods offer:
+   * Faster iterations for development and testing
+   * Less bloat accumulation
+   * Easier bug resolution
+     
+While longer iteration is useful for:
+   * Longer testing windows
+   * Running applications that require longer running histories
+   * A more stable network environment
+     
 
 #### Execution client
 
+Given a known period and current timestamp, client can always calculate the number of lifecycle iterations from genesis 0 and create a new genesis with latest parameters.
+
+For example, let the current state of the network be: 
+    
+    ChainId: 39438088
+    Genesis Zero Timestamp: 1691694000 (Mon Nov 14 19:00:00 UTC 2022)
+    Current Timestamp: 1692036000 (Mon Aug 14 19:00:00 UTC 2023)
+    Period: 604800 seconds (1 week)
+
 * Number of iterations:
-    *  `i` = `int((latest timestamp` - `genesis_0.timestamp) / period)`
+    *  `i` = `int((current_timestamp` - `genesis_0_timestamp) / period)`
+    *  `i` = `int((1692036000 - 1668448800) / 604800)`
+    *  `i` = `39`
 * Timestamp of current genesis:
-    * `genesis_timestamp` = `period` * `i` + `genesis_0.timestamp`
+    * `genesis_timestamp` = `period` * `i` + `genesis_0_timestamp`
+    * `genesis_timestamp` = `604800` * `39` + `1668448800`
+    * `genesis_timestamo` = `1692036000`
 * Current EL ChainId:
-    * `chainId` = `genesis_0.chainId` + `i`
+    * `chainId` = `genesis_0_chainId` + `i`
+    * `chainId` = `39438088 + 39`
+    * `chainId` = `39438127`
+ 
+Staying in sync with rapidly changing genesis values like this will introduce new challenges to clients like:
+   * Handling sync frequency
+   * Managing data persistence
+   * User experience
+   * Infrastructure management
+   
+Mitigating this challenges will require solutions like:
+   * Automatic updates that minimize the need for manual intervention.
+   * Graceful transitions that minimize netowrk disruptions during resets.
+   * Flexible configurations that allows users set how often they want to check for updates
+   * Clear documentation that helps users understand the need for frequenct changes and how to effectively manage them.
+   * A user notification system, like a simple countdown showing how much time before the upcoming network reset.
+   
 
 #### Consensus client
 
-Genesis generation in CL client uses the same parameters as EL but requires also updated genesis state in ssz. 
+Genesis generation in CL client uses the same parameters as EL, while also requiring the generation of the updated genesis state in SSZ format, with minor changes which include:
+
+* A deposit (smart) contract responsible for managing the process of validators making deposits to participate in the network.
```

Comment:
The deposit contract is on EL and doesn't change, CL just needs to know its address, deployment hight etc which stay constant